### PR TITLE
Adding UUID for jti generation

### PR DESCRIPTION
**Pull request by djw8605**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -6,6 +6,7 @@ set( CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake )
 
 find_package( jwt-cpp REQUIRED )
 find_package( CURL REQUIRED )
+find_package(UUID REQUIRED)
```
Comment (bbockelm) [inline CMakeLists.txt]:
```suggestion
find_package( UUID REQUIRED )
```

(just to match whitespace of previous lines)

File: CMakeLists.txt [hunk 2]
```diff
@@ -34,10 +35,10 @@ endif()
 
 pkg_check_modules(SQLITE REQUIRED sqlite3)
 
-include_directories( "${PROJECT_SOURCE_DIR}" ${JWT_CPP_INCLUDES} ${CURL_INCLUDES} ${LIBCRYPTO_INCLUDE_DIRS} ${SQLITE_INCLUDE_DIRS})
+include_directories( "${PROJECT_SOURCE_DIR}" ${JWT_CPP_INCLUDES} ${CURL_INCLUDES} ${LIBCRYPTO_INCLUDE_DIRS} ${SQLITE_INCLUDE_DIRS}  ${UUID_INCLUDE_DIRS})
```
Comment (bbockelm) [inline CMakeLists.txt]:
```suggestion
include_directories( "${PROJECT_SOURCE_DIR}" ${JWT_CPP_INCLUDES} ${CURL_INCLUDES} ${LIBCRYPTO_INCLUDE_DIRS} ${SQLITE_INCLUDE_DIRS}  ${UUID_INCLUDE_DIRS} )
```

File: cmake/FindUUID.cmake [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  else (UUID_FOUND)
+    if (UUID_FIND_REQUIRED)
+      message(FATAL_ERROR "Could not find UUID")
+    endif (UUID_FIND_REQUIRED)
+  endif (UUID_FOUND)
+
+  # show the UUID_INCLUDE_DIRS and UUID_LIBRARIES variables only in the advanced view
+  mark_as_advanced(UUID_INCLUDE_DIRS UUID_LIBRARIES)
+
+endif (UUID_SYMBOL)
```
Comment (bbockelm) [inline cmake/FindUUID.cmake]:
```suggestion
endif (UUID_SYMBOL)
```

(add EOL -- not sure if GitHub suggestion interface adds this correctly...)

Comment (djw8605) [reply cmake/FindUUID.cmake]: It doesn't, but I added the new line.

**Pull request merged**